src/Security/Authentication/Negotiate/src/LdapSettings.cs

```diff
@@ -56,6 +57,19 @@ public class LdapSettings
         /// </summary>
         public LdapConnection LdapConnection { get; set; }
 
+        /// <summary>
+        /// The expiration that should be used for entries in the cache for user claims, defaults to 10 minutes.
+        /// This is a sliding expiration that will extend each time claims for a user is retrieved.
+        /// </summary>
+        public TimeSpan ClaimsCacheEntryExpiration { get; set; } = TimeSpan.FromMinutes(10);
+
+        /// <summary>
+        /// How many user claim results to store in the cache, defaults to 1024.
+        /// </summary>
+        public int ClaimsCacheSize { get; set; } = 1024;
+
+        internal MemoryCache ClaimsCache { get; set; }
```

Comment:
This is different from the certificate validation cache in that it's not stored in DI. It might be a bit odd to have the cache in LdapSettings but I'm not sure if there's any value in putting this cache in DI.

Reply:
Main value of putting it in DI is if you want to make it easy for them to plug in their own cache implementation

Reply:
While that's true, I'm not sure we'd want that kind of flexibility here at the moment. I don't see it as particularly useful in this scenario.

Reply:
That's fine, I was just pointing out what advantages putting it in DI has